scipy/stats/_morestats.py

```diff
@@ -4114,5 +4165,6 @@ def directionalmean(samples, *, axis=0, normalize=True):
         vectornorms = np.linalg.norm(samples, axis=-1, keepdims=True)
         samples = samples/vectornorms
     mean = np.mean(samples, axis=0)
-    directional_mean = mean / np.linalg.norm(mean, axis=-1, keepdims=True)
-    return directional_mean
+    magnitude = np.linalg.norm(mean, axis=-1, keepdims=True)
+    directional_mean = mean / magnitude
+    return DirectionalStats(directional_mean, magnitude.squeeze(-1))
```

Comment:
```suggestion
    return DirectionalStats(directional_mean, magnitude.squeeze(-1)[()])
```

to avoid scalar arrays being returned.

Reply:
Done.